ui/pages/send/send-content/send-amount-row/amount-max-button/amount-max-button.component.js

```diff
@@ -1,81 +1,47 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import React from 'react';
 import classnames from 'classnames';
-
-export default class AmountMaxButton extends Component {
-  static propTypes = {
-    balance: PropTypes.string,
-    buttonDataLoading: PropTypes.bool,
-    clearMaxAmount: PropTypes.func,
-    inError: PropTypes.bool,
-    gasTotal: PropTypes.string,
-    maxModeOn: PropTypes.bool,
-    sendToken: PropTypes.object,
-    setAmountToMax: PropTypes.func,
-    setMaxModeTo: PropTypes.func,
-    tokenBalance: PropTypes.string,
-  };
-
-  static contextTypes = {
-    t: PropTypes.func,
-    metricsEvent: PropTypes.func,
+import { useDispatch, useSelector } from 'react-redux';
+import { getBasicGasEstimateLoadingStatus } from '../../../../../selectors';
+import {
+  getSendMaxModeState,
+  isSendFormInvalid,
+  toggleSendMaxMode,
+} from '../../../../../ducks/send';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import { useMetricEvent } from '../../../../../hooks/useMetricEvent';
+
+export default function AmountMaxButton() {
+  const buttonDataLoading = useSelector(getBasicGasEstimateLoadingStatus);
+  const inError = useSelector(isSendFormInvalid);
```

Comment:
Should we be using `isSendFormInError` here rather than `isSendFormInvalid`? The variable name kinda implies it should be the `InError` one. Though `isSendFormInvalid` covers a few more seemingly-relevant cases like gas being set below the minimum.

Not sure the value in having these two separate selectors, now that I'm thinking about it :thinking:

Reply:
I can probably rewrite some code to remove the isSendFormInError one.

Reply:
https://github.com/MetaMask/metamask-extension/pull/10965/commits/29bd524b3dfac27ea4a5e3e5a44099f5d8236ba6

Reply:
oh and the variable renamed in: https://github.com/MetaMask/metamask-extension/pull/10965/commits/1b5931cf04b3d582740a6f5c28ecee1c03297e26

Reply:
Nice, that works!